features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SessionReplayConfiguration.kt

```diff
@@ -31,10 +31,12 @@ data class SessionReplayConfiguration internal constructor(
      * A Builder class for a [SessionReplayConfiguration].
      * @param sampleRate must be a value between 0 and 100. A value of 0
      * means no session will be recorded, 100 means all sessions will be recorded.
+     * If this value is not provided then Session Replay will default to a 100 sample rate.
      */
-    class Builder(@FloatRange(from = 0.0, to = 100.0) private val sampleRate: Float) {
+    class Builder(@FloatRange(from = 0.0, to = 100.0) sampleRate: Float? = null) {
```

Comment:
**Question:** why not just add default value in constructor like:
```
 class Builder(@FloatRange(from = 0.0, to = 100.0) sampleRate: Float? = SAMPLE_IN_ALL_SESSIONS)
```

Reply:
I would have done this, but I want to avoid having the property be visible externally as optional. By doing it this way, externally the property of sampleRate is seen as `Float` instead of `Float?`

Reply:
> [!NOTE]
> You can still have the property be non null (although, as the property is private it doesn't really matter), and have the constructor parameter use the default value : 

```
    class Builder(
        @FloatRange(from = 0.0, to = 100.0) private val sampleRate: Float = SAMPLE_IN_ALL_SESSIONS
    ) {
       // …
    }
```